src/ooxml/java/org/apache/poi/xssf/streaming/reader/StreamedCell.java

```diff
@@ -0,0 +1,578 @@
+/* ====================================================================
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+package org.apache.poi.xssf.streaming.reader;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import org.apache.poi.ss.formula.FormulaParseException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Comment;
+import org.apache.poi.ss.usermodel.Hyperlink;
+import org.apache.poi.ss.usermodel.RichTextString;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.util.CellAddress;
+import org.apache.poi.ss.util.CellRangeAddress;
+import org.apache.poi.util.NotImplemented;
+
+/**
+ * Represents cell in a row Value of cell is represented as a string.
+ *
+ */
+public class StreamedCell implements Cell {
+    private String value;
+    private int cellNumber;
+    private CellType cellType;
+
+    /**
+     * <pre>
+     * Return cell value
+     * </pre>
+     * 
+     * Return the value of a cell in String format. Value will be same as how it
+     * is represented in excel.
+     * 
+     * @return String
+     */
+    public String getValue() {
+        return value;
+    }
+
+    public void setValue(String value) {

```

Comment:
is this Cell able to store non-string values?
There's a couple ways full data type support could be implemented.
1. Define a StringCell, NumberCell, DateCell, FormulaCell, ErrorCell, BlankCell classes with the ability to convert between the classes if the user wants to change the type of cell.
2. Make `value` an Object
3. Make Cell have the fields stringValue, numberValue, dateValue, formulaValue, errorValue, only one of which is valid at any one time, determined by whatever is stored in cellType.